gnd/src/androidTest/java/com/google/android/gnd/AddFeatureTest.java

```diff
@@ -0,0 +1,152 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd;
+
+import static androidx.test.espresso.Espresso.onData;
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isCompletelyDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.not;
+
+import android.content.Context;
+import androidx.room.Room;
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import com.google.android.gnd.persistence.local.LocalDatabaseModule;
+import com.google.android.gnd.persistence.local.room.LocalDatabase;
+import com.google.android.gnd.persistence.remote.FakeRemoteDataStore;
+import com.google.android.gnd.persistence.remote.FakeRemoteStorageManager;
+import com.google.android.gnd.persistence.remote.RemoteDataStore;
+import com.google.android.gnd.persistence.remote.RemoteStorageManager;
+import com.google.android.gnd.persistence.remote.RemoteStorageModule;
+import com.google.android.gnd.persistence.uuid.FakeUuidGenerator;
+import com.google.android.gnd.persistence.uuid.OfflineUuidGenerator;
+import com.google.android.gnd.system.auth.AuthenticationManager;
+import com.google.android.gnd.system.auth.AuthenticationModule;
+import com.google.android.gnd.system.auth.FakeAuthenticationManager;
+import dagger.Binds;
+import dagger.Module;
+import dagger.Provides;
+import dagger.hilt.InstallIn;
+import dagger.hilt.android.components.ApplicationComponent;
+import dagger.hilt.android.qualifiers.ApplicationContext;
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+import dagger.hilt.android.testing.UninstallModules;
+import javax.inject.Singleton;
+import org.junit.Rule;
+import org.junit.Test;
+
+@UninstallModules({AuthenticationModule.class, RemoteStorageModule.class,
+  LocalDatabaseModule.class})
+@HiltAndroidTest
+public class AddFeatureTest {
+
+  // Ensures that the Hilt component is initialized before running the ActivityScenarioRule.
+  public @Rule(order = 0)
+  HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+  // Sets the preferences so no login is required and an active project is selected.
+  public @Rule(order = 1)
+  SetPreferencesRule preferencesRule = new SetPreferencesRule();
+
+  public @Rule(order = 2)
+  ActivityScenarioRule scenarioRule =
+    new ActivityScenarioRule(MainActivity.class);
+
+  @Module
+  @InstallIn(ApplicationComponent.class)
+  abstract class TestDependenciesModule {
```

Comment:
I don't think this is super idiomatic. I would move these modules to their own files or put them at the bottom of this file. I've opened this file to read the tests, not to read the setup. The only reason you would put a class inside another class is if you need to access the parent from the child.